scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1540,13 +1552,32 @@ public String getTaskServerLogsByTag(String jobId, String taskTag)
         if (myJobsOnly) {
             user = ident.getUsername();
         }
-        return dbManager.getJobs(offset,
-                                 limit,
-                                 user,
-                                 filterCriteria.isPending(),
-                                 filterCriteria.isRunning(),
-                                 filterCriteria.isFinished(),
-                                 sortParameters);
+
+        Page<JobInfo> jobsInfo = dbManager.getJobs(offset,
+                                                   limit,
+                                                   user,
+                                                   filterCriteria.isPending(),
+                                                   filterCriteria.isRunning(),
+                                                   filterCriteria.isFinished(),
+                                                   sortParameters);
+
+        /**
+         * Add/inject to each JobInfo the list of signals, if they exist, used by the job.
+         */
+        try {
+            List<String> jobHavingSignalsIds = new ArrayList<String>(publicStore.keySet(SIGNAL_ORIGINATOR,
+                                                                                        SIGNAL_TASK_ID,
+                                                                                        signalsChannel));
+            jobsInfo.getList()
+                    .stream()
+                    .filter(jobInfo -> jobHavingSignalsIds.contains(jobInfo.getJobId().value()))
+                    .map(jobInfo -> insertJobSignals(jobInfo))
+                    .collect(Collectors.toList());
+        } catch (InvalidChannelException e) {
```

Comment:
we can catch another kind of exceptions here so why only InvalidChannelException ?

Reply:
It is the only type of exceptions that can be thrown